### DISCUSSION TITLE
🛠️ Project ideas to build with MACI

### DISCUSSION THREAD

**Discussion by andyguzmaneth** [privacy-ethereum/maci#1136] [Ideas]
_This is a compendium of ideas to build with/for/to MACI. Ideal place to brainstorm and collect ideas of this that are cool-to-build but haven't been build yet. There is some overlap with [MACI 2024 Product Roadmap](https://github.com/privacy-scaling-explorations/maci/discussions/859) where indicated_ 

----------------------

### 📱Apps Ideas
- QF Deployer: similar to [Conduit](https://conduit.xyz/) for L2s, create a website so that anyone can create a MACI/QF round and chose it's configuration with a few clicks
- ~~Frontends:~~
     ~~- to manage MACI rounds (i.e. an admin portal for coordinators to operate rounds)~~
     ~~- to contribute to MACI rounds (i.e. voting applications for users)~~
- Continue Quadratic Dollar Homepage ([QDH](https://github.com/privacy-scaling-explorations/qdh))
- ~~Subgraph for MACI: this will aid frontend development and data querying ([maci-subgraph](https://ethglobal.com/showcase/maci-subgraph-89h7c))~~
- Quadratic Funding App (like [clr.fund](https://github.com/clrfund/monorepo/))
- RetroPGF App (like [maci-platform](https://github.com/privacy-scaling-explorations/maci-platform/))
- DAO Voting Governance App
    - Ensure voice credits can be safely assigned based on a token/erc721 balance (keep in mind flashloans) 
- “Movie night” voting app (or any voting app) with reusable frontend components for other MACI enabled apps
- Extract some of the MACI primitives to build something new!
- Plugins for Anon Polls in:
    - Telegram
    - Discord
    - Slack
 - e2e [MACI Platform](https://github.com/privacy-scaling-explorations/maci-platform) interface with Account Abstraction features:
    - Smart wallets with passkeys
    - Paymaster for sponsoring TXs  
 - Recursive MACI:
    - Tournament style using MACI 
    - Start a voting round based on the result of the previous one
    - Provide a clean interface for configuration/usage  

### 🧩Integrations
- MACI <> Anon Aadhaar
- MACI <> Proof of Passport / ZK Passport
- MACI <> Giveth
- MACI <> Devfolio

### ⚙️Protocol Improvements & Research ideas
- ~~Build additional [Gatekeeper contracts](https://github.com/privacy-scaling-explorations/maci/tree/dev/contracts/contracts/gatekeepers) for MACI user registration (like for [EAS](https://github.com/privacy-scaling-explorations/maci/pull/1109) and [Hats Protocol](https://github.com/privacy-scaling-explorations/maci/pull/1191))~~
- Account Abstraction explorations - make user onboarding easier and gas-free:
    - Paymasters
        - Custom paymaster which allows only voting and signup to MACI 
    - Seedless wallets 
        - Recovery methods with zk-email or simil 
- Explore "Future areas of research & development" [MACI 2024 Product Roadmap](https://github.com/privacy-scaling-explorations/maci/discussions/859) but some ideas include:
    - Gass-less MACI
        - Implement a relayer that submits messages and signups on chain for users 
    - Reducing coordinator trust assumptions
    - Gas efficiency
    - Other ZK-scaling techniques (like folding)
    - Different voting strategies (weighted, group matching)
        - Customise the Tally circuit to allow for different voting types (e.g. implement a cap on final votes etc.) 
- Writing circuits in different DSLs like: 
    - Noir
    - Leo
    - Python with Chiquito
- Anti-collusion detection using machine learning algorithms

### 📊 Benchmarks 
- Complete benchmarks of the system

### 🔧Tooling & Developer Experience
- 🐳📦 **Create a Docker container to quickly spin-up a MACI instance locally**
- Tool to generate Circom docs from comments to properly document the circuits
- Intuitive quadratic voting/funding React component for [MACI Platform](https://github.com/privacy-scaling-explorations/maci-platform) to simplify UI/UX for QV votes

**Comment by 0xarcano** [comment]
Hello @aguzmant103, I'm interested in the MACI Dockerization idea, please guide me in the process, should I create a repository or work in a MACI repo fork?

**Reply by Gelois0** [reply]
Also, I am interested. Can we make a squad ? @prodasl

**Reply by ctrlc03** [reply]
Hey @prodasl and @Gelois0 thanks for your interest to work on this. We'd love for you to contribute to MACI. In this case, you could work on a fork and submit a PR with the docker files - does that sound good?

**Reply by 0xarcano** [reply]
Hello @ctrlc03 thank you for the response, I think a good way is to create the fork and put a new folder in the root of the project named Docker to organize all the Docker related files in it, I'll start it soon and coordinate with @Gelois0 to start working on it. 
Thanks!

**Reply by ctrlc03** [reply]
@prodasl awesome lmk if we can help you in any way!
